### ISSUE TITLE
The "Self::" paths aren't properly processed in the doc-comments.

### ISSUE THREAD

**Issue by iddm** [rust-lang/rust-analyzer#15439]
Given this code:

```rust
struct A;
/// Bla bla - [`Self::process_messages`].
pub fn new() -> Self {
    Self
}

/// Documentation for processing messages.
pub fn process_messages(&self) {}
```

The `Self::process_messages` is never a hyperlink and doesn't navigate anywhere, it is not highlighted. But in the generated output of `cargo doc` (the HTML pages), it is properly highlighted and navigates to the correct place within the code.

If I change the `Self` to `A` in this example, everything works, but this is rarely the intention, considering how useful the `Self` alias is.

Is it possible to solve this in the rust-analyzer?

rust-analyzer version: 0.3.1615-standalone
rustc 1.71.1 (eb26296b5 2023-08-03)

**Comment by lnicola** [comment]
@lowr is this the FIXME [here](https://github.com/rust-lang/rust-analyzer/pull/15405/files#diff-d55e974dbc3fcd02151c5e71130f7b065dfea3935a192298e6b475e5409415deR233)?

**Comment by Veykril** [comment]
No, I think it's more the FIXME [here](https://github.com/rust-lang/rust-analyzer/pull/15405/files#diff-d55e974dbc3fcd02151c5e71130f7b065dfea3935a192298e6b475e5409415deR279-R280)
